shell/common/gin_helper/object_template_builder.h

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2019 GitHub, Inc. All rights reserved.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#ifndef SHELL_COMMON_GIN_HELPER_OBJECT_TEMPLATE_BUILDER_H_
+#define SHELL_COMMON_GIN_HELPER_OBJECT_TEMPLATE_BUILDER_H_
+
+#include "shell/common/gin_helper/function_template.h"
+
+namespace gin_helper {
+
+// This class works like gin::ObjectTemplateBuilder, but operates on existing
+// prototype template instead of creating a new one.
+//
+// It also uses gin_helper::CreateFunctionTemplate for function templates to
+// support gin_helper types.
+class ObjectTemplateBuilder {
```

Comment:
Maybe as a follow up we can add this constructor to upstream gin, so that class can reused without duplicating ?

Reply:
This class also relies on our fork of `function_template.h` so simply patching gin would not be enough, I'll add comment for it.